### PR TITLE
Initial commit of ISO versioning work.

### PR DESCRIPTION
**Pull request by dlorenc**
This adds code to fetch the ISO URL from GCS. This also adds an initial versionmap.json file.

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -40,6 +41,30 @@ var (
...
+	ISOURL string
+}
+
+type isoMap map[string]version
+
+func GetIsoUrl(mapURL, v string) (string, error) {
+	// Download the version map file.
+	resp, err := http.Get(mapURL)
+	if err != nil {
+		return "", nil
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: Why not return err here?

File: pkg/minikube/cluster/cluster.go [hunk 2]
```diff
@@ -40,6 +41,30 @@ var (
 	certs = []string{"apiserver.crt", "apiserver.key"}
 )
 
+type version struct {
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: The ISO-specific code here should maybe be in a separate file.

File: cmd/minikube/cmd/start.go [hunk 3]
```diff
@@ -43,15 +43,25 @@ var (
 	minikubeISO string
 )
 
+func getConfig() cluster.MachineConfig {
+	if minikubeISO != "" {
+		return cluster.MachineConfig{MinikubeISO: minikubeISO}
+	}
+	isoURL, err := cluster.GetIsoUrl(constants.ISOMapURL, constants.ISOVersion)
+	if err != nil {
```
Comment (luxas) [inline cmd/minikube/cmd/start.go]: Maybe check for `err != nil || isoURL == ""` just to be safe

**Comment by dlorenc** [comment]
Actually let's hold off on this for now. I'll send a simpler change and we can take another look later.

**Pull request closed without merging**